File: Hash Tables/Easy/C++/3005. Count Elements With Maximum Frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        
        int maxFreq = 0;
        unordered_map<int, int> m;
        int count = 0;

        for(int i: nums){
            m[i]++;
            maxFreq = max(maxFreq, m[i]);
        }

        for(int i: nums){
            if(m[i] == maxFreq){
                count++;
            }
        }

        return count;
    }
};

File: Hash Tables/Easy/3005. Count Elements With Maximum Frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {

        // Time Complexity : O(n)

        // Space Complexity : O(n)
        
        
        Map<Integer, Integer> map = new HashMap<>();
        int max = 0, count = 0;

        for(Integer i: nums){
            if(!map.containsKey(i)){
                map.put(i, 1);
                if(1 > max){
                    max = 1;
                    count = 1;
                }
                else if(max == 1){
                    count++;
                }
            }
            else{
                map.put(i, map.get(i) + 1);
                if(map.get(i) > max){
                    max = map.get(i);
                    count = 1;
                }
                else if(map.get(i) == max){
                    count++;
                }
            }
        }

        return max * count;

    }
}
